### ISSUE TITLE
sauceOptions.setCapturePerformance() should force or encourage a setting of test name

### ISSUE THREAD

**Issue by nadvolod** [saucelabs/simple_sauce#241]
Currently, if we do this:
```
        sauceOptions = new SauceOptions();
        sauceOptions.setExtendedDebugging(true);
        sauceOptions.setCapturePerformance(true);
```
The outcome of the performance test will look like [this](https://snipboard.io/5N98dD.jpg). This does make sense because how can you capture history of a test that has a unique identifier as a name.
To fix this issue currently we need to do:
```
        sauceOptions.setCapturePerformance(true);
        sauceOptions.setName("simplePerformanceTest");
```
My recommendation is to update the API to:
```
sauceOptions.setCapturePerformance("simplePerformanceTest");
```
This solves the problem above. Furthermore, it solves another problem of needing to set true flag for `setCapturePerformance()` which I believe is redundant. `setCapturePerformance()` is disabled by default.  Hence, if we set it, the only option left is to enable it. Providing the user the option to `setCapturePerformance(false)` is silly I believe

**Comment by titusfortner** [comment]
Yeah, but that the parameter is the test name isn't obvious. I'd really like to see support for bindings for each test runner that will automatically grab the test name somehow. (So an automatic default similar to what we're doing for build name). But I haven't looked into how difficult this would be for JUnit/TestNG

**Comment by nadvolod** [comment]
@titusfortner why do you say it's not obvious as the method signature will show `setCapturePerformance(String testName)`? Furthermore, this is currently a usability issue and potentially a bug because the only way to enable performance is to do
```
        sauceOptions.setCapturePerformance(true);
        sauceOptions.setName("simplePerformanceTest"); 
```
However, this isn't obvious and isn't documented anywhere... I encountered this because I tried to use it

**Comment by titusfortner** [comment]
I don't like relying on an IDE to tell you the parameters; theoretically, good code should have obvious arguments.
My solution to this was the to throw a custom error if setting it before setting a name. 
This will make it easy for us to implement a default naming strategy going forward without people needing to change anything.

**Comment by titusfortner** [comment]
well, I guess they would still be setting the name earlier, either way. /shrug

**Comment by nadvolod** [comment]
I understand what you mean about good code and obvious arguments. @titusfortner do you feel that using legacy Sauce API and Sauce Bindings that our API is obvious?

**Current Solution (1)**
```java
        sauceOptions.setCapturePerformance(true);
//not mandatory to set user name... But will misbehave if not set
        sauceOptions.setName("simplePerformanceTest");
```
So instead of weirdly forcing users to use 2 methods, why can't we just use one method with one parameter? Isn't this what method parameters are for... If you don't like the previous method name, maybe we can do?

**Option 2**
```java
sauceOptions.enableFrontEndPerformance("simplePerformanceTest");
```
**Option 3**
Or you force the user to always set test name and then do
```java
//no boolean as front end perf is disabled by default
sauceOptions.setFrontEndPerformance();
```
I'm not as big on forcing the user to always set test name. 

**Option 4**
Maybe we can automatically set the test name ourselves if it's not set and then use `sauceOptions.setFrontEndPerformance();`

**Option 5**
```java
        sauceOptions.setCapturePerformance(true);
//throw runtime exception if method below isn't used
        sauceOptions.setName("simplePerformanceTest");
```
